lib/providers/order/uniswapxService.ts

```diff
@@ -14,23 +16,41 @@ export class UniswapXServiceProvider implements OrderServiceProvider {
     this.log = _log.child({ quoter: 'UniswapXOrderService' });
   }
 
-  async postOrder(order: Order, signature: string, quoteId?: string): Promise<void> {
+  async postOrder(order: Order, signature: string, quoteId?: string): Promise<ErrorResponse | void> {
     this.log.info({ orderHash: order.hash() }, 'Posting order to UniswapX Service');
 
     const axiosConfig = {
       timeout: ORDER_SERVICE_TIMEOUT_MS,
     };
-    await axios.post(
-      `${this.uniswapxServiceUrl}dutch-auction/order`,
-      {
-        encodedOrder: order.serialize(),
-        signature: signature,
-        chainId: order.chainId,
-        quoteId: quoteId,
-        orderType: V2_ORDER_TYPE,
-      },
-      axiosConfig
-    );
-    this.log.info({ orderHash: order.hash() }, 'Order posted to UniswapX Service');
+    try {
+      await axios.post(
+        `${this.uniswapxServiceUrl}dutch-auction/order`,
+        {
+          encodedOrder: order.serialize(),
+          signature: signature,
+          chainId: order.chainId,
+          quoteId: quoteId,
+          orderType: V2_ORDER_TYPE,
+        },
+        axiosConfig
+      );
+      this.log.info({ orderHash: order.hash() }, 'Order posted to UniswapX Service');
+    } catch (e) {
+      if (e instanceof AxiosError) {
+        this.log.error({ error: e.response?.data }, 'Error posting order to UniswapX Service');
+        return {
+          statusCode: e.response?.data.statusCode,
```

Comment:
```suggestion
          statusCode: e.response?.data?.statusCode,
```

Reply:
and same below

Reply:
`data` isn't optional so should be fine?